privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ToggleReportsDataStore.kt

```diff
@@ -0,0 +1,234 @@
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+import kotlinx.coroutines.flow.first
+import timber.log.Timber
+
+interface ToggleReportsDataStore {
+
+    suspend fun storeDismissLogicEnabled(dismissLogicEnabled: Boolean)
+
+    suspend fun getDismissLogicEnabled(): Boolean
+
+    suspend fun storePromptLimitLogicEnabled(promptLimitLogicEnabled: Boolean)
+
+    suspend fun getPromptLimitLogicEnabled(): Boolean
+
+    suspend fun storeDismissInterval(dismissInterval: Int)
+
+    suspend fun getDismissInterval(): Int
+
+    suspend fun storePromptInterval(promptInterval: Int)
+
+    suspend fun getPromptInterval(): Int
+
+    suspend fun storeMaxPromptCount(maxPromptCount: Int)
+
+    suspend fun getMaxPromptCount(): Int
+
+    suspend fun insertTogglePromptDismiss()
+
+    suspend fun getReportsSent(): List<String>
+
+    suspend fun insertTogglePromptSend()
+
+    suspend fun getPromptsDismissed(): List<String>
+
+    suspend fun lastPromptWasAccepted(): Boolean
+
+    suspend fun setLastPromptWasAccepted(accepted: Boolean)
+
+    suspend fun canPrompt(): Boolean
+}
+
+@ContributesBinding(AppScope::class)
+@SingleInstanceIn(AppScope::class)
+class SharedPreferencesToggleReportsDataStore @Inject constructor(
+    @ToggleReports private val store: DataStore<Preferences>,
+    // @AppCoroutineScope private val appCoroutineScope: CoroutineScope,
+) : ToggleReportsDataStore {
+
+    private object Keys {
+        val TOGGLE_REPORTS_PROMPT_LIMIT_LOGIC_ENABLED = booleanPreferencesKey(name = "TOGGLE_REPORTS_PROMPT_LIMIT_LOGIC_ENABLED")
+        val TOGGLE_REPORTS_DISMISS_LOGIC_ENABLED = booleanPreferencesKey(name = "TOGGLE_REPORTS_DISMISS_LOGIC_ENABLED")
+        val TOGGLE_REPORTS_SENT = stringSetPreferencesKey(name = "TOGGLE_REPORTS_SENT")
+        val TOGGLE_REPORTS_PROMPTS_DISMISSED = stringSetPreferencesKey(name = "TOGGLE_REPORTS_PROMPTS_DISMISSED")
+        val TOGGLE_REPORTS_DISMISS_INTERVAL = intPreferencesKey(name = "TOGGLE_REPORTS_DISMISS_INTERVAL")
+        val TOGGLE_REPORTS_PROMPT_INTERVAL = intPreferencesKey(name = "TOGGLE_REPORTS_PROMPT_INTERVAL")
+        val TOGGLE_REPORTS_MAX_PROMPT_COUNT = intPreferencesKey(name = "TOGGLE_REPORTS_MAX_PROMPT_COUNT")
+        val TOGGLE_REPORTS_LAST_PROMPT_WAS_ACCEPTED = booleanPreferencesKey(name = "TOGGLE_REPORTS_LAST_PROMPT_WAS_ACCEPTED")
+    }
+
+    private companion object {
+        const val DEFAULT_INTERVAL = 172800
+        const val DEFAULT_MAX_PROMPT = 3
+    }
+
+    override suspend fun storeDismissLogicEnabled(dismissLogicEnabled: Boolean) {
+        store.edit { prefs ->
+            prefs[TOGGLE_REPORTS_DISMISS_LOGIC_ENABLED] = dismissLogicEnabled
+        }
+    }
+
+    override suspend fun getDismissLogicEnabled(): Boolean {
+        return store.data.first()[TOGGLE_REPORTS_DISMISS_LOGIC_ENABLED] ?: false
```

Comment:
This will throw an exception if there's no element, should we use `firstOrNull` instead?

Reply:
We don't need to store these anymore, but I updated the two remaining usages.